src/protocols/secure_channel/CheckinMessage.h

```diff
@@ -74,13 +74,16 @@ class DLL_EXPORT CheckinMessage
      * @param[in]       hmac128KeyHandle  Key handle with which to verify the received nonce in the check-in payload (using HMAC).
      * @param[in]       payload           The received payload to decrypt and parse
      * @param[out]      counter           The counter value retrieved from the payload
+     *                                    If an error occurs, no value will be set.
      * @param[in,out]   appData           The optional application data decrypted. The input size of appData must be at least the
-     * size of GetAppDataSize(payload) + sizeof(CounterType), because appData is used as a work buffer for the decryption process.
-     * The output size on success will be GetAppDataSize(payload).
+     *                                    size of GetAppDataSize(payload) + sizeof(CounterType), because appData is used as a work
+     * buffer for the decryption process. The output size on success will be GetAppDataSize(payload).
+     * If an error occurs, appData might countain data, but the data CANNOT be used since we were not able to validate it.
      *
      * @return CHIP_ERROR_INVALID_MESSAGE_LENGTH if the payload is shorter than the minimum payload size
      *         CHIP_ERROR_BUFFER_TOO_SMALL if appData buffer is too small
      *         CHIP_ERROR_INVALID_ARGUMENT if the provided arguments cannot be used to parse the Check-In message
```

Comment:
It seems that the test is not currently testing for CHIP_ERROR_INVALID_ARGUMENT, can this error still be generated?

Reply:
removed since no longuer applicable